src/BrowserProtocol.js

```diff
@@ -0,0 +1,94 @@
+import off from 'dom-helpers/events/off';
+import on from 'dom-helpers/events/on';
+import invariant from 'invariant';
+
+import createPath from './utils/createPath';
+
+export default class BrowserProtocol {
```

Comment:
The idea is that a Redux user would make a store enhancer with:

```js
const historyEnhancer = createHistoryEnhancer(
  new BrowserProtocol(),
  queryMiddleware,
);

const store = createStore(
  reducer,
  compose(historyEnhancer, /* ... */),
);

store.dispatch(FarceActions.init());
const history = createStoreHistory(store, state => state.location);
```

This way, the only API exposed is the protocol and any location enhancer middlewares. Everything else is hidden from the user.

Down the road, this will allow something like:

```js
const store = createStore(
  reducer,
  compose(
    applyMiddleware(createNamedRoutesMiddleware(routes)),
    historyEnhancer,
    // ...,
  )
);
```